### ISSUE TITLE
[Bug?] `LinkHints.activateHover` does not detect GitHub Code Navigation unless the keyword is hovered with mouse once

### ISSUE THREAD

**Issue by MahdiNazemi** [gdh1995/vimium-c#909]
**Describe the bug**

I have recently started using `LinkHints.activateHover` and find it very helpful. However, it looks like it does not capture the GitHub Code Navigation links unless the keywords are hovered on with the mouse.

Sample code:
https://github.com/motokimura/yolo_v1_pytorch/blob/master/detect.py#L242

1. Initial link hints when the mouse is not hovered on any keyword:
    <img width="1695" alt="Screenshot 2023-03-31 at 10 57 42 AM" src="https://user-images.githubusercontent.com/11276349/229204132-9fc3aedf-852a-46f8-859a-858d82ea8642.png">

2. Link hints when some keywords are hovered on:
    <img width="1695" alt="Screenshot 2023-03-31 at 10 57 55 AM" src="https://user-images.githubusercontent.com/11276349/229204393-3ee829fd-aa6f-4f64-b8af-c61811e961f3.png">

3. Link hints when a few more keywords are hovered on:
    <img width="1695" alt="Screenshot 2023-03-31 at 10 58 05 AM" src="https://user-images.githubusercontent.com/11276349/229204441-9665e0f9-6596-4022-a7eb-0c0b5638bd1f.png">

**Browser, OS, and Vimium C versions**

- Browser name: Firefox
- Browser version: 111.0.1
- Vimium C version: 1.99.97
- OS name and version: macOS Ventura 13.3

**Comment by MahdiNazemi** [comment]
If I define the following mappings
```
map <c-f> LinkHints.activateHover clickableOnHost="\\bgithub##.pl-s1;"
map f LinkHints.activate clickableOnHost="\\bgithub##.pl-s1;"
```
and do a hover or click, followed by a click, I can see the code navigation pop-up.

I assume the first hover or click executes the javascript that generates a 
```html
<span class="pl-token" ...>...</span>
```
and the following click opens the pop-up.

**Comment by MahdiNazemi** [comment]
I tried to use the answer provided in the following comment to define a mapping that automatically clicks on a link hint after hovering over it.

https://github.com/gdh1995/vimium-c/issues/891#issuecomment-1483854629:
> ```vim
> map F LinkHints.activateHover $then="$D+mid_click"
> run <v-mid_click> sim_mid#mousedown+$D+sim_mid#mouseup+$D+lh_open
> run <v-sim_mid> dispatchEvent#button=1&type=$s mask
> map <v-lh_open> LinkHints.click mode="incognito" incognito=false direct="hovered"
> ```

What I defined looks like this:
```vim
map <a-c-m-s-f> LinkHints.activateHover $then="$D+left_click" clickableOnHost="\\bgithub##.pl-s1,.pl-en;"
run <v-left_click> sim_left#mousedown+$D+sim_left#mouseup+$D+lh_open
run <v-sim_left> dispatchEvent#button=0&type=$s mask
map <v-lh_open> LinkHints.click
```

However, this does not open the pop-up.

1. Am I using `lh_open` correctly?
2. Do I need to introduce a delay between hover and click to allow GitHub's javascript to load the `<span>` element?
3. Is there an easier when to implement hover followed by click?

**Comment by gdh1995** [comment]
But, `direct=hovered` only selects a last `.pl-s1` element, but not the newly added `.pl-token` element, so the click fails.

There's no working around in Vimium C v1.99.97, so I'll add a new feature to let you select a child of a hovered element.

**Comment by MahdiNazemi** [comment]
> But, `direct=hovered` only selects a last `.pl-s1` element, but not the newly added `.pl-token` element, so the click fails.

> If I define the following mappings
> 
> ```
> map <c-f> LinkHints.activateHover clickableOnHost="\\bgithub##.pl-s1;"
> map f LinkHints.activate clickableOnHost="\\bgithub##.pl-s1;"
> ```
> 
> and do a hover or click, followed by a click, I can see the code navigation pop-up.

How come a second click on the same link hint shows the pop-up? Does it automatically propagate the click to child elements?

> There's no working around in Vimium C v1.99.97, so I'll add a new feature to let you select a child of a hovered element.

I spent a lot of time reading the docs and issues last night and suspect `directOptions` may help me solve this issue until a new version is released. I think [this comment](https://github.com/gdh1995/vimium-c/issues/581#issuecomment-1072422844) has a useful example.

**Comment by gdh1995** [comment]
> a second click on the same link

No, when in a second LinkHints, the hint you selected with a same name is actually different - it points to `.pl-token` instead of `.pl-s1`, so a manual click "works".

But the new mappings will hover `.pl-s1` if such a element hasn't been clicked, and `LinkHints.click#direct=hovered` can not select a child now.

**Comment by MahdiNazemi** [comment]
> But, `direct=hovered` only selects a last `.pl-s1` element, but not the newly added `.pl-token` element, so the click fails.
> 
> There's no working around in Vimium C v1.99.97, so I'll add a new feature to let you select a child of a hovered element.

@gdh1995, Would you please explain how to define a mapping for this feature in v1.99.99?

**Comment by gdh1995** [comment]
Such a mapping should help:
```
map xxx LinkHints.activateHover clickableOnHost="\\bgithub##.pl-s1;" then={".pl-s1":"pl_token"}
run <v-pl_token> 800wait+LinkHints.click#direct="hovered=.pl-token"
```

If you prefer shorter lines, you may write:
```
map xxx LinkHints.activateHover \
  clickableOnHost="\\bgithub##.pl-s1;" \
  then={".pl-s1":"pl_token"}
run <v-pl_token> 800wait+\\
  LinkHints.click#direct="hovered=.pl-token"
```

**Comment by MahdiNazemi** [comment]
Thanks a lot! Works like a charm!

Here is another version with more selectors:
```vim
map <a-c-m-s-f> LinkHints.activateHover \
  clickableOnHost="\\bgithub##.pl-s1,.pl-en,.pl-v;" \
  then={".pl-s1,.pl-en,.pl-v":"pl_token"}
run <v-pl_token> 800wait+\\
  LinkHints.click#direct="hovered=.pl-token"
```
